### ISSUE TITLE
[Accessibility]: Cannot navigate to link to create repository filters

### ISSUE THREAD

**Issue by efritz** [sourcegraph/sourcegraph#35187]
### Audit type

Screen reader navigation

### User journey audit issue

#33518

### Problem description

<img width="884" alt="impossible to nav" src="https://user-images.githubusercontent.com/103087/167503897-098a8ecc-4f2d-4ad9-b9cf-926686401d85.png">

It's currently impossible using VoiceOver to select this link to create a new input field. It currently only reads the leading text, then allows us to skip to the period after the link. Not sure how to navigate properly here.

### Expected behavior

I can clicky clicky that link and it should add additional dynamic forms.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
